### ISSUE TITLE
[Feature Request] Allow hydra.main to work with Kedro 

### ISSUE THREAD

**Issue by neltacigreb** [facebookresearch/hydra#2063]
# 🚀 Feature Request

Kedro is an open-source Python framework for creating reproducible, maintainable and modular data science code.
Its main feature is to create data pipelines that are defined by chaining nodes containing the data transformation and model manipulation steps.
In my opinion, this is like DBT but for the data science workflow.
Project description here : https://github.com/kedro-org/kedro 

When trying to use Hydra via the hydra.main decorator, the conf folder path points to something unreachable.

## Motivation

Would be nice to be able to combine these 2 awesome packages :)

### Reproducing issue 

python version: 3.8.12
kedro version: 0.17.7
hydra version: 1.1.1

The bug appears when trying to set hydra.main decorator on the main function used by Kedro, namely `register_pipelines()`. Steps to reproduce :  

- Setup the iris_dataset toy project by following those steps : https://kedro.readthedocs.io/en/stable/02_get_started/05_example_project.html
- Add the files required by Hydra in the conf folder (config.yaml and base/master.yaml). 
Rewrite existing files extension (yml->yaml)
- Add the hydra.main decorator in src/[package_name]/pipeline_registry.py : 
<pre>
<b>import hydra</b>
<b>@hydra.main(config_path="../../conf", config_name="config")</b>
def register_pipelines(cfg: DictConfig) -> Dict[str, Pipeline]:
    """Register the project's pipelines.

    Returns:
        A mapping from a pipeline name to a ``Pipeline`` object.

    """

    data_engineering_pipeline = de.create_pipeline()
    data_science_pipeline = ds.create_pipeline()

    return {
        "de": data_engineering_pipeline,
        "ds": data_science_pipeline,
        "__default__": data_engineering_pipeline + data_science_pipeline,
    }
</pre>

This will result in the following error: 
```
Primary config module 'get_started.conf' not found.
Check that it's correct and contains an __init__.py file
```
_note: get_started is the name of the package in /src_

### Cause of the issue

After some digging, it appears that the configuration path resolved by hydra.main does not exist. 
- When the bug appears, calling `self.get_sources()` while being in `ConfigLoaderImpl.ensure_main_config_source_available` returns this : `[provider=hydra, path=pkg://hydra.conf, provider=main, path=pkg://conf, provider=schema, path=structured://]`
- It should actually be this : `[provider=hydra, path=pkg://hydra.conf, provider=main, path=file:///PATH_TO_PROJECT//conf, provider=schema, path=structured://]`
It appears that Hydra doesn't know how to get `file:///PATH_TO_PROJECT`, and replaces it by `pkg://`

## Pitch

Not really sure how to solve and which library should be adapted to correct this bug, so I wrote a [similar post on Kedro's issues tracker](https://github.com/kedro-org/kedro/issues/1303). 
A Kedro project executes by calling `kedro run` from a terminal and not by calling the script manually, so I guess it comes from somewhere here.

Right now i'm using a workaround by generating the conf via `initialize()` and `compose()` : 
<pre>
<b>from hydra import compose, initialize</b>

def register_pipelines() -> Dict[str, Pipeline]:
    """Register the project's pipelines.

    Returns:
        A mapping from a pipeline name to a ``Pipeline`` object.

    """
    <b>initialize(config_path="../../conf")</b>
    <b>cfg = compose(config_name="config")</b>

    data_engineering_pipeline = de.create_pipeline()
    data_science_pipeline = ds.create_pipeline()

    return {
        "de": data_engineering_pipeline,
        "ds": data_science_pipeline,
        "__default__": data_engineering_pipeline + data_science_pipeline,
    }
</pre>

## Follow up question
This also raises the question on how to do conf overrides from the command line. It's possible to pass a YAML configuration file to load the run command arguments from, via the `--config` argument of `kedro run`, but I haven't tested it yet.

tl;dr: hydra.main() called in a unusual way, leading to the impossibility to find the conf folder.

**Comment by Jasha10** [comment]
Hi @neltacigreb,
Could you please upload a .zip file with the files from your reproducible example?

**Comment by neltacigreb** [comment]
Hi @Jasha10 , here is the example I described : https://github.com/neltacigreb/kedroXhydra
It's been created by running `kedro new --starter=pandas-iris`, then a few lines tweaked to implement Hydra.

**Comment by neltacigreb** [comment]
Solved in https://github.com/kedro-org/kedro/issues/1303

**Comment by neltacigreb** [comment]
Some update on this, I managed to make a custom decorator to be used on the register_pipelines() function, and that will reproduce most of the behavior of the hydra.main() decorator.
It works, but is very crafty. So I don't know if it will always have the same behavior as the original decorator.

[Documentation and source can be found in this repo](https://github.com/neltacigreb/kedroXhydra/blob/hook_implem_nodec/src/get_started/hooks.py)
